Стать ментором

askar0103 opened the pull request.

askar0103 committed:
Стать ментором. Вторая попытка разрешения конфликта: выполнен rebase с разрешением конфликтов, force push

Mike-GB22 commented on src/main/java/school/faang/user_service/mapper/mentorship/MentorshipMapper.java:
```diff
@@ -0,0 +1,27 @@
+
+import java.util.List;
+
+@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
```
Виталий советовал использовать ReportingPolicy.ERROR

askar0103 replied:
Спасибо

Mike-GB22 commented:
я весь код посмотрел, не вижу за что может у меня глаз зацепиться, так что я бы апрув поставил

Nafanuil commented on src/main/java/school/faang/user_service/controller/mentorship/MentorshipController.java:
```diff
@@ -0,0 +1,39 @@
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/mentorship")
+public class MentorshipController {
```
очень лаконичный и красивый класс, давай только вынесем сюда мапинг из сервиса, все таки чем больше ты отделяешь слои друг от друга тем чище и проще код.

askar0103 replied:
Спасибо) Вынес маппинг из сервиса сюда

Nafanuil commented on src/main/java/school/faang/user_service/service/mentorship/MentorshipService.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.List;
+
+@Service
+@Validated
```
так же предлагаю вынести валидацию на уровень контроллера, все по той де причине, в сервисе должна оставаться только бизнес логика, а заодно предлагаю попробовать написать любую кастомную аннотацию валидации чтобы понять как они работают (к примеру можешь сделать свою аннjтацию MyPositive)

askar0103 replied:
Спасибо большущее! вынес валидацию на уровень контроллера, сделал кастомную аннотацию @ValidId

askar0103 committed:
Стать ментором.

feat:
- Добавлена кастомная аннотация @ValidId для валидации ID
- Реализован валидатор ValidLongIdValidator

askar0103 committed:
Стать ментором. Вторая попытка разрешения конфликта
